libs/permission-checker/src/Check/JobQueue/CanRunJob.php

```diff
@@ -30,24 +31,34 @@ public function checkPermissions(StorageApiToken $token): void
         }
 
         if ($token->hasFeature(Feature::PROTECTED_DEFAULT_BRANCH)) {
-            $this->checkProtectedDefaultBranch($token->getRole());
+            $this->checkProtectedDefaultBranch($token);
         } elseif ($token->isRole(Role::READ_ONLY)) {
             throw PermissionDeniedException::roleDenied($token->getRole(), 'run jobs');
         }
     }
 
-    private function checkProtectedDefaultBranch(Role $role): void
+    private function checkProtectedDefaultBranch(StorageApiToken $token): void
     {
-        $isAllowed = match ($role) {
+        $isAllowed = match ($token->getRole()) {
             Role::PRODUCTION_MANAGER => $this->branchType === BranchType::DEFAULT,
             Role::DEVELOPER, Role::REVIEWER => $this->branchType !== BranchType::DEFAULT,
+            Role::NONE => $token->hasPermission(TokenPermission::CAN_RUN_JOBS),
```

Comment:
Non-admin token s oprávněním canRunJobs v soxovém projektu

Reply:
Takovej blbej nezasvecenej dotaz - jako admin token muzu poustet joby, i kdyz mam `canRunJobs: false`? Jako mozna chapu, jak se k tomu doslo, ale kdyz na to kouknu bez kontextu, tak WTF :smile:

Reply:
Ano je to tak, viz https://keboola.atlassian.net/wiki/spaces/ENGG/pages/2992799745/PENDING+RFC-2023-005+SOX+Privileged+tokens#Variant-C "else if the API request token has canCreateJobs flag"

Reply:
Jo muzes. To spolu nesouvisi tohle permission.